pkg/controller/che/che_controller.go

```diff
@@ -1191,6 +1155,179 @@ func (r *ReconcileChe) Reconcile(request reconcile.Request) (reconcile.Result, e
 	return reconcile.Result{}, nil
 }
 
+// creates `che` serviceaccount with needed permissions
+func createServiceAccounts(instance *orgv1.CheCluster, r *ReconcileChe) error {
+	// create service accounts:
+	// che is the one which token is used to create workspace objects
+	cheServiceAccount := deploy.NewServiceAccount(instance, "che")
+	if err := r.CreateServiceAccount(instance, cheServiceAccount); err != nil {
+		return err
+	}
+
+	if !instance.Spec.Auth.OpenShiftoAuth {
+
+		if instance.Namespace == instance.Spec.Server.WorkspaceNamespaceDefault {
+			// this role is needed to manage che-workspace serviceaccount inside che namespace
+			cheRole := deploy.NewRole(instance, "che", []rbac.PolicyRule{
+				{
+					APIGroups: []string{"authorization.openshift.io", "rbac.authorization.k8s.io"},
+					Resources: []string{"roles"},
+					Verbs:     []string{"get", "create"},
+				},
+				{
+					APIGroups: []string{"authorization.openshift.io", "rbac.authorization.k8s.io"},
+					Resources: []string{"rolebindings"},
+					Verbs:     []string{"get", "update", "create"},
+				},
+			})
+			if err := r.CreateNewRole(instance, cheRole); err != nil {
+				return err
+			}
+			cheExecrolesRoleBinding := deploy.NewRoleBinding(instance, "che", cheServiceAccount.Name, cheRole.Name, "Role")
+			if err := r.CreateNewRoleBinding(instance, cheExecrolesRoleBinding); err != nil {
+				return err
+			}
+
+			// create RoleBindings for created (and existing ClusterRole) roles and service accounts
+			cheRoleBinding := deploy.NewRoleBinding(instance, "che-edit", cheServiceAccount.Name, "edit", "ClusterRole")
+			if err := r.CreateNewRoleBinding(instance, cheRoleBinding); err != nil {
+				return err
+			}
+		} else {
+			cheCreateNamespacesName := fmt.Sprintf(cheCreateNamespaces, instance.Namespace)
+			cheCreateNameSpacesRole := deploy.NewClusterRole(instance, cheCreateNamespacesName, []rbac.PolicyRule{
+				{
+					APIGroups: []string{"project.openshift.io"},
+					Resources: []string{"projectrequests"},
+					Verbs: []string{"create", "update"},
+				},
+				{
+					APIGroups: []string{""},
+					Resources: []string{"namespaces"},
+					Verbs: []string{"create", "update"},
+				},
+			})
+			if err := r.CreateNewClusterRole(instance, cheCreateNameSpacesRole); err != nil {
+				return err
+			}
+
+			// this binding is needed to create new namespaces for workspaces
+			// `${namespace}-clusterrole-create-namespaces` ClusterRole should be created during che-operator deploy
```

Comment:
What do you mean ? That it is just created 5 lines above ?

Reply:
We delegate some part permissions to the che server

Reply:
OK